lib/ansible/modules/system/systemd.py

```diff
@@ -392,17 +397,26 @@ def main():
             module.fail_json(msg='failure %d during daemon-reexec: %s' % (rc, err))
 
     if unit:
+        systemctl_unit = systemctl
         found = False
         is_initd = sysv_exists(unit)
         is_systemd = False
 
+        if module.params['root']:
+            systemctl_unit += " --root=%s" % module.params['root']
+
+        # if scope is 'system' or None, we can ignore as there is no extra switch.
+        # The other choices match the corresponding switch
+        if module.params['scope'] not in (None, 'system'):
+            systemctl_unit += " --%s" % module.params['scope']
+
         # check service data, cannot error out on rc as it changes across versions, assume not found
-        (rc, out, err) = module.run_command("%s show '%s'" % (systemctl, unit))
+        (rc, out, err) = module.run_command("%s show '%s'" % (systemctl_unit, unit))
```

Comment:
fail if `root` is used:
show seams not to take into account the --root parameter, so we are "showing" the wrong unit.
```
max@mde-oxalide % sudo systemctl --root /var/lib/machines/arch-64-tmp/ show org.cups.cupsd.service | grep LoadState
LoadState=loaded
max@mde-oxalide % sudo find /var/lib/machines/arch-64-tmp/ -name org.cups.cupsd.service
max@mde-oxalide %
```